File: src/my_project/easy_problems/from201to250/reformat_date.py
from typing import List 

class Solution:
    def reformatDate(self, date: str) -> str:
        
        temp_list = date.split()
        months = ["Jan", "Feb", "Mar", "Apr", "May", "Jun", "Jul", "Aug", "Sep", "Oct", "Nov", "Dec"]
        temp_dict = dict()
        for i in range(12):
            temp_dict[months[i]] = i+1
    
        answer = ""
        
        day = temp_list[0]
        month = temp_list[1]
        year = temp_list[2]
        
        answer = f'{year}-{temp_dict[month]:0>2d}-{int(day[0:len(day)-2]):0>2d}'
        
        return answer
    
    
solution = Solution()

print(solution.reformatDate(date = "20th Oct 2052"))

print(solution.reformatDate(date = "6th Jun 1933"))

print(solution.reformatDate(date = "26th May 1960"))

File: my_project_typescript/easy_problems_typescript/from101to150/reformatDate.ts
function reformatDate(date: string): string {

    let tempList = date.split(" ")
    let months: Array<string> = ["Jan", "Feb", "Mar", "Apr", "May", "Jun", "Jul", "Aug", "Sep", "Oct", "Nov", "Dec"]
    let tempDict: any = {}

    for (let i = 0; i < 12; i++){
        tempDict[months[i]] = i+1
    }

    function padLeadingZeros(num:number, size:number): string {
        let s = num.toString() +"";
        while (s.length < size) s = "0" + s;
        return s;
    }


    let answer: string = ""
    let day: string = tempList[0]
    let month: string = tempList[1]
    let year: string = tempList[2]
    
    

    answer = year + "-" + padLeadingZeros(tempDict[month],2) + "-"+  padLeadingZeros(parseInt(day.slice(0,day.length-2)),2)

    return answer 


};


console.log(reformatDate("20th Oct 2052"))

console.log(reformatDate("6th Jun 1933"))

console.log(reformatDate("26th May 1960"))
